node/src/test/scala/com/wavesplatform/database/LevelDBWriterSpec.scala

```diff
@@ -408,5 +409,18 @@ class LevelDBWriterSpec extends FreeSpec with Matchers with TransactionGen with
       }
     }
 
+    "dont parse irrelevant transactions in transferById" in {
+      baseTest(time => preconditions(time.correctedTime())) { (writer, _) =>
```

Comment:
There's actually no need to use `baseTest` here. You could just use `db.put` and then instantiate `LevelDBWriter` manually. This will also allow to keep `readWrite` method private.
It may also be a good idea to use a generator to test random transaction IDs.